v3.6/security/kubernetes-network-policy.md

```diff
@@ -0,0 +1,178 @@
+
+#### Ingress and egress
+
+The bulk of securing network connections typically revolves around defining egress and ingress rules. From the point of view of a Kubernetes pod, **ingress** is incoming connections to the pod, and **egress** is outgoing connections from the pod. In Kubernetes network policy, you create ingress and egress “allow” rules independently (egress, ingress, or both). 
+
+#### Default deny/allow behavior
+
+**Default allow** means all connections are allowed by default, unless otherwise specified. 
+**Default deny** means all connections are denied by default, unless explicitly allowed. 
+
+### How to
+
+Before you create your first network policy, you need to understand the default network policy behaviors. If no Kubernetes network policies apply to a pod, then all connections to/from the pod are allowed (default-allow). As a result, if you do not create any network policies, then all pods are allowed to communicate freely with all other pods. If one or more Kubernetes network policies apply to a pod, then only the connections specifically defined in that network policy are allowed (default-deny).
+
+You are now ready to start fine-tuning connections that should be allowed. 
+
+- [Create ingress policies](#create-ingress-policies)
+- [Allow ingress connections from pods in the same namespace](#allow-ingress-connections-from-pods-in-the-same-namespace)
+- [Allow ingress connections from pods in a different namespace](#allow-ingress-connections-from-pods-in-a-different-namespace)
+- [Create egress policies](#create-egress-policies)
+- [Allow ingress connections from pods in the same namespace](#allow-egress-connections-from-pods-in-the-same-namespace)
+- [Allow egress connections to IP addresses or CIDR range](#allow-egress-connections-to-ip-addresses-or-cidr-range)
+- [Best practice: create deny-all default network policy](#best-practice-create-deny-all-default-network-policy)
+- [Create deny-all default ingress and egress network policy](#create-deny-all-default-ingress-and-egress-network-policy)
+
+#### Create ingress policies
+
+Create ingress network policies to allow inbound connections from other pods. 
+
+Network policies apply to pods within a specific **namespace**. Policies can include one or more ingress rules. To specify which pods in the namespace the network policy applies to, use a **pod selector**. Within the ingress rule, use another pod selector to define which pods allow incoming connections, and the **ports** field to define on which ports connections are allowed. 
+
+##### Allow ingress connections from pods in the same namespace
+
+In the following example, incoming connections to pods with label **color=blue** are allowed only if they come from a pod with **color=red**, on port **80**.
+
+<pre>
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-same-namespace
+  namespace: default
+spec:
+  podSelector:
+    matchLabels: {“<b>color</b>”: “<b>blue</b>”}
```

Comment:
While this is a valid networkpolicy yaml, we usually show dictionaries (`{}`)  in a different way, like this:

```
spec:
  podSelector:
    matchLabels:
      <b>color</b>: <b>blue</b>
```

The quotes `"` aren't necessary either.

They both are interpreted the same way, so not a huge deal either way. I'd argue that my suggested way looks a little nicer...